echopype/mask/api.py

```diff
@@ -493,23 +506,34 @@ def frequency_differencing(
         chanA = chanAB[0]
         chanB = chanAB[1]
 
-    # get the left-hand side of condition
-    lhs = source_Sv["Sv"].sel(channel=chanA) - source_Sv["Sv"].sel(channel=chanB)
-
-    # create mask using operator lookup table
-    da = xr.where(str2ops[operator](lhs, diff), True, False)
+    # If Sv data is not dask array
+    if not isinstance(source_Sv["Sv"].variable._data, dask.array.Array):
+        da = _frequency_diff_and_mask(source_Sv["Sv"], chanA, chanB, diff)
+    # If Sv data is dask array
+    else:
+        # Get the final data array template
+        template = source_Sv["Sv"].isel(channel=0).drop_vars("channel")
+        # Iterate over all the chunks
+        da = source_Sv["Sv"].map_blocks(
```

Comment:
@anantmittal I've updated your code to use the built-in xarray's `map_blocks` to map over the blocks, which will return the template data array. This is exactly the same way like you had, but now we don't need to convert to dask array, then convert back 😄 I just realized that this function exists!